File: src/main/java/application/history/history/application/services/GetAllHistoriesService.java
package application.history.history.application.services;

import application.history.history.domain.entities.History;
import application.history.history.domain.repositories.HistoryRepository;
import org.springframework.stereotype.Service;

@Service
public class GetAllHistoriesService {
    private final HistoryRepository repository;

    public GetAllHistoriesService(HistoryRepository repository) {
        this.repository = repository;
    }

    public Iterable<History> run() {
        return this.repository.findAll();
    }
}

File: src/history/main/kotlin/com/example/history/application/services/GetAllHistoriesService.kt
package com.example.history.application.services

import com.example.history.domain.entities.History
import com.example.history.domain.repositories.HistoryRepository
import org.springframework.stereotype.Service

@Service
class GetAllHistoriesService(repository: HistoryRepository) {
    private val repository: HistoryRepository

    fun run(): Iterable<History> {
        return repository.findAll()
    }

    init {
        this.repository = repository
    }
}
